projects/plugins/jetpack/extensions/blocks/button/button.php

```diff
@@ -171,6 +171,10 @@ function get_button_styles( $attributes ) {
 	$has_typography_styles       = array_key_exists( 'style', $attributes ) && array_key_exists( 'typography', $attributes['style'] );
 	$has_custom_font_size        = $has_typography_styles && array_key_exists( 'fontSize', $attributes['style']['typography'] );
 	$has_custom_text_transform   = $has_typography_styles && array_key_exists( 'textTransform', $attributes['style']['typography'] );
+	$has_named_border_color      = array_key_exists( 'borderColor', $attributes );
+	$has_custom_border_color     = array_key_exists( 'style', $attributes ) && array_key_exists( 'border', $attributes['style'] ) && array_key_exists( 'color', $attributes['style']['border'] );
+	$has_border_style            = array_key_exists( 'style', $attributes ) && array_key_exists( 'border', $attributes['style'] ) && array_key_exists( 'style', $attributes['style']['border'] );
+	$has_border_width            = array_key_exists( 'style', $attributes ) && array_key_exists( 'border', $attributes['style'] ) && array_key_exists( 'width', $attributes['style']['border'] );
```

Comment:
The border block support also [allows "per-side" or split borders](https://github.com/WordPress/gutenberg/blob/trunk/lib/block-supports/border.php#L102-L112) e.g. border-top, border-left etc.

The style engine's get styles function will convert a style object to the appropriate CSS classes and inline styles. Here's core's version, [wp_style_engine_get_styles](https://github.com/WordPress/wordpress-develop/blob/bc3fdb4648d850b03d9a2e12cffdbe43cd91741b/src/wp-includes/style-engine.php#L63), and [Gutenberg's](https://github.com/WordPress/gutenberg/blob/trunk/packages/style-engine/style-engine.php#L41).

An example of its use can be found in the border block support linked above. The results are effectively the same as those on the JS side via `useBorderProps`.

Reply:
I've updated to add styles for each individual side, based on the Gutenberg border supports you linked above.